drivers/npmx/npmx_shell.c

```diff
@@ -64,14 +58,189 @@ typedef enum {
+	bool bvalue; /* bool result. */
+} shell_arg_result_t;
+
+/** @brief Shell argument structure. */
+typedef struct {
+	const shell_arg_type_t type; /* Shell argument type. */
+	const char *name; /* Shell argument name. */
+	shell_arg_result_t result; /* Shell argument result. */
+} shell_arg_t;
+
+/** @brief Shell arguments infomation structure. */
+typedef struct {
+	const int expected_args; /* Expected arguments count. */
+	shell_arg_t arg[SHELL_ARG_MAX_COUNT]; /* Shell arguments. */
+} args_info_t;
 
 static const struct device *pmic_dev = DEVICE_DT_GET(DT_NODELABEL(npm_0));
 
+static void print_set_error(const struct shell *shell, const char *str)
+{
+	shell_error(shell, "Error: unable to set %s.", str);
+}
+
+static void print_get_error(const struct shell *shell, const char *str)
+{
+	shell_error(shell, "Error: unable to get %s.", str);
+}
+
+static void print_convert_error(const struct shell *shell, const char *src, const char *dst)
+{
+	shell_error(shell, "Error: unable to convert %s to %s.", src, dst);
+}
+
+static void print_ivalue(const struct shell *shell, int32_t value)
+{
+	shell_print(shell, "Value: %d.", value);
+}
+
+static void print_uvalue(const struct shell *shell, uint32_t value)
+{
+	shell_print(shell, "Value: %u.", value);
+}
+
+static void print_uvalue_unit(const struct shell *shell, uint32_t value, const char *unit)
+{
+	shell_print(shell, "Value: %u %s.", value, unit);
+}
+
+static void print_ivalue_unit(const struct shell *shell, int32_t value, const char *unit)
+{
+	shell_print(shell, "Value: %d %s.", value, unit);
+}
+
+static void print_bsuccess(const struct shell *shell, bool value)
+{
+	shell_print(shell, "Success: %d.", value);
+}
+
+static void print_usuccess_unit(const struct shell *shell, uint32_t value, const char *unit)
+{
+	shell_print(shell, "Success: %u %s.", value, unit);
+}
+
+static void print_isuccess_unit(const struct shell *shell, int32_t value, const char *unit)
+{
+	shell_print(shell, "Success: %d %s.", value, unit);
+}
+
+static void print_isuccess(const struct shell *shell, int32_t value)
+{
+	shell_print(shell, "Success: %d.", value);
+}
+
+static void print_usuccess(const struct shell *shell, uint32_t value)
+{
+	shell_print(shell, "Success: %u.", value);
+}
+
+static void print_bvalue(const struct shell *shell, bool value)
+{
+	shell_print(shell, "Value: %d.", value);
+}
+
+static void print_help(const struct shell *shell, int32_t index, const char *str)
+{
+	shell_print(shell, "%d-%s", index, str);
+}
+
+static const char *int_name_get(shell_arg_type_t arg_type)
+{
+	switch (arg_type) {
+	case SHELL_ARG_TYPE_INT_VALUE:
+	case SHELL_ARG_TYPE_UINT_VALUE:
+	case SHELL_ARG_TYPE_BOOL_VALUE:
+		return " value";
+	case SHELL_ARG_TYPE_UINT_INDEX:
+		return " instance index";
+	default:
+		return "";
+	}
+}
+
+static bool arguments_check(const struct shell *shell, size_t argc, char **argv,
+			    args_info_t *args_info)
+{
+	assert(args_info);
+	assert(args_info->expected_args < SHELL_ARG_MAX_COUNT);
```

Comment:
Will fail for `buck_gpio_set`. `SHELL_ARG_MAX_COUNT` should be equal to 4, or it should be `<=` instead of `<` here.